src/prefer-filter-first.ts

```diff
@@ -0,0 +1,44 @@
+import { Rule } from "eslint";
+
+export const preferFilterFirstRule: Rule.RuleModule = {
```

Comment:
I think it would make sense to slightly rearchitecture a rule like this in a way where we could generate rules like these. Currently, this will only capture arr.map.filter calls, but the same optimization could apply to arr.reduce.filter or arr.sort.filter. There are plenty of inefficiencies around iteration that this could help uncover. In addition, it would also be nice if this rule could support instances where the order might not strictly be .map.filter and for example also catch something like .map.sort.filter

Reply:
I like that idea, didn't think of it. I had a look into the source of https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/prefer-array-flat-map.js, which looks much less error-prone and also seems like it's easier to add autofix. Do you think it might make sense to contribute to the unicorn plugin instead? (since it has an established release flow, readme, etc.)